### PR TITLE
Refactored all usages of directories used by datanode and managed opensearch

### PR DESCRIPTION
**Pull request by todvora**
Refactor all usages and pointers of datanode and managed-opensearch directories. All are now handled on one place only (DatanodeDirectories class), properly resolved, **using nodeId as a subdir** to avoid configuration collisions. Added javadoc for all methods to make clear what are our expectations and usages of these directories.

Additionally we now check the presence and permissions to each of the directories in advance, during preflight checks. This should reveal any possible problems significantly sooner than during runtime later.

/nocl

## How Has This Been Tested?
Added unit tests, existing set of integration tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: data-node/src/main/java/org/graylog/datanode/OpensearchDistribution.java [hunk 1]
```diff
@@ -28,4 +29,7 @@ public OpensearchDistribution(Path path, String version) {
         this(path, version, null, null);
     }
 
+    public Path getOpensearchBinFile() {
```
Comment (janheise) [inline data-node/src/main/java/org/graylog/datanode/OpensearchDistribution.java:32]: Maybe name this `getOpenSearchBinDir` ? as with most other paths or really `File`?

Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/OpensearchDistribution.java:32]: This is really the executable bin file. Maybe renaming to getOpensearchExecutable?

**Review by janheise — APPROVED** [review]
LGTM!

File: data-node/src/main/java/org/graylog/datanode/OpensearchDistribution.java (hunk 1 above)
Comment (todvora) [reply data-node/src/main/java/org/graylog/datanode/OpensearchDistribution.java]: Done, thx!

**Pull request merged**